staging/src/k8s.io/client-go/tools/cache/store.go

```diff
@@ -99,20 +100,38 @@ type ExplicitKey string
 // The key uses the format <namespace>/<name> unless <namespace> is empty, then
 // it's just <name>.
 //
-// TODO: replace key-as-string with a key-as-struct so that this
-// packing/unpacking won't be necessary.
+// Clients that want a structured alternative can use ObjectToName or MetaObjectToName.
+// Note: this would not be a client that wants a key for a Store because those are
+// necessarily strings.
+//
+// TODO maybe some day?: change Store to be keyed differently
 func MetaNamespaceKeyFunc(obj interface{}) (string, error) {
```

Comment:
I would like to templetize the things that use this such that we can use a struct{name, namespace string} as a key directly.

Reply:
Oh, I misunderstood what you were suggesting templating (I thought you meant workqueue entries).  Both would be interesting, IMHO.

Reply:
Yeah I meant both, sorry! At least eventually.

I'm a little under the weather today, I'll come back to this PR soon hopefully